extensions/stackdriver/stackdriver.cc

```diff
@@ -364,9 +363,13 @@ void StackdriverRootContext::record() {
   ::Extensions::Stackdriver::Metric::record(
       outbound, local_node, peer_node, request_info,
       !config_.disable_http_size_metrics());
-  if (enableServerAccessLog() && shouldLogThisRequest(request_info)) {
+  if ((enableAllAccessLog() ||
+       (enableClientAccessLogOnError() &&
```

Comment:
I still don't think we need to distinguish client or not. This can just be enableAccessLogOnError. Direction is already distinguished at filter configuration.

Reply:
We are saying that we enable access log on error on client side only... thus the distinction...
I think the code will work for server side too if we remove the distinction.. would be a good add to server side logging... @mandarjog, @douglas-reid  wdyt?

Reply:
I think “logOnError” will suffice. 
We will not set it in server side logs config for now, but will work just as well.

Reply:
Done!